esp/services/ws_esdlconfig/ws_esdlconfigservice.hpp

```diff
@@ -57,17 +57,26 @@ class CWsESDLConfigEx : public CWsESDLConfig
 
     bool attachServiceToDali() override
     {
+        SpinBlock b(m_daliDetachSpinLock);
```

Comment:
I don't see the point of this spinlock (and spinlocks are in general a bad idea).

It's not preventing any race other than two threads writing the member at the same time with different values, in which case the value written would be (indeterminately) one of the values that the two threads tried to write. But the same is true with the spinlock in place - the value written will be (indeterminately) one of the two values.

If you believe that the processor might somehow write a boolean value that is nether true nor false should two threads try to write it at the same time, you could replace the member with a std::atomic<bool>. But there's still a race, in that if two threads both call this function at the same time, you don't know what state will win. If that's a significant race, it needs addressing outside this function.

Reply:
spinlock chosen to avoid context switches (critsec/mutex) since the synchronized logic is inexpensive and the same thread doesn't attempt to re-enter...

The calls to attachServiceToDali and detach are serialized at the ESPcfg level

The intent is to synchronize the methods that read the flag with the setting of the flag. All the onXXX methods that require a dali connection reject requests when detached from dali. 

The worst case scenario if I don't synch, we could accept a dali connection dependent request after we've declared to be detached.

Reply:
@rpastrana The concern you described is true, but the spin lock won't prevent it. To prevent Dali being detached while operation is in place, you have to synchronize the detach and the functions that use Dali connection. Another way is to use the canDeatchFromDali method/variable. Setting and getting of the "can" variable should be together and synchronized. You can create an object that checks and turns off "can" in constructor, and turns on "can" in destructor, then you can just declare a variable of this type at the beginning of each ws_esdlconfig function.

Reply:
@rpastrana the spin lock won't prevent the concern you described. Imagine inside onGetESDLDefinition, the call isDetachedFromDali() returns false, and your function proceeds. But before it completes, detachServiceFromDali() gets called, and it succeeds. But nothing prevents the onGetESDLDefinition method from continuing executing. To address the concern completely the detach and the whole onXXX methods need to be synchronized. Maybe a ReadWrite lock or a use counter?

Reply:
@mayx the issue you describe is different (albeit a legitimate issue worth addressing).
The current change prevents processing of incoming requests as the state is in flux. 
IMHO the issue you describe ultimately needs to be handled in the CEsdlSDSStore where the actual DALI connections are managed. I'll open a jira for that work and will discuss with you further.